tests/platform/api/test_watchdog.py

```diff
@@ -0,0 +1,176 @@
+            test_config.update(test_config[platform]['default'])
+
+        if platform in test_config and hwsku in test_config[platform]:
+            test_config.update(test_config[platform][hwsku])
+
+        return test_config
+
+
+    def test_arm_disarm_states(self, testbed_devices, platform_api_conn, conf):
+        ''' arm watchdog with a valid timeout value, verify it is in armed state,
+        disarm watchdog and verify it is in disarmed state
+        '''
+
+        duthost = testbed_devices['dut']
+        localhost = testbed_devices['localhost']
+
+        test_timeout = conf['valid_timeout']
+        actual_timeout = watchdog.arm(platform_api_conn, test_timeout)
+
+        assert actual_timeout != -1
+        assert actual_timeout >= test_timeout
+        assert watchdog.is_armed(platform_api_conn)
+
+        assert watchdog.disarm(platform_api_conn)
+        assert not watchdog.is_armed(platform_api_conn)
+
+        res = localhost.wait_for(host=duthost.hostname,
+                port=22, state="stopped", delay=5, timeout=test_timeout,
+                module_ignore_errors=True)
+
+        assert 'exception' in res
+
+    def test_remaining_time(self, duthost, platform_api_conn, conf):
+        ''' arm watchdog with a valid timeout and verify that remaining time API works correctly '''
+
+        test_timeout = conf['valid_timeout']
+
+        assert watchdog.get_remaining_time(platform_api_conn) == -1
+
+        actual_timeout = watchdog.arm(platform_api_conn, test_timeout)
+        remaining_time = watchdog.get_remaining_time(platform_api_conn)
+
+        assert remaining_time > 0
+        assert remaining_time <= actual_timeout
+
+        remaining_time = watchdog.get_remaining_time(platform_api_conn)
+        time.sleep(2)
+        assert watchdog.get_remaining_time(platform_api_conn) < remaining_time
+
+    def test_periodic_arm(self, duthost, platform_api_conn, conf):
+        ''' arm watchdog several times as watchdog deamon would and verify API behaves correctly '''
+
+        test_timeout = conf['valid_timeout']
+        actual_timeout = watchdog.arm(platform_api_conn, test_timeout)
+        time.sleep(2)
+        remaining_time = watchdog.get_remaining_time(platform_api_conn)
+        actual_timeout_new = watchdog.arm(platform_api_conn, test_timeout)
+
+        assert actual_timeout == actual_timeout_new
+        assert watchdog.get_remaining_time(platform_api_conn) > remaining_time
+
+    def test_arm_different_timeout_greater(self, duthost, platform_api_conn, conf):
+        ''' arm the watchdog with greater timeout value and verify new timeout was accepted;
+        If platform accepts only single valid timeout value, @greater_timeout should be None.
+        '''
+
+        test_timeout = conf['valid_timeout']
+        test_timeout_greater = conf.get('greater_timeout', None)
+        if test_timeout_greater is None:
+            pytest.skip('"greater_timeout" parameter is required for this test case')
```

Comment:
If 'greater_timeout' is None the test will be skipped - e.g. platform does not support setting different timeouts. It is up to the user to modify watchdog.yml for specific platform.
Otherwise, 'greater_timeout' has a default value defined in watchdog.yml.

The confusion probably comes from the line:

```python
test_timeout_greater = conf.get('greater_timeout', None)
```
'greater_timeout' key will be available in any case, so I'll do the change